wasm/types.go

```diff
@@ -11,6 +11,16 @@ import (
 	"github.com/go-interpreter/wagon/wasm/leb128"
 )
 
+type Marshaller interface {
```

Comment:
I know it's not the proper spelling but I'd be inclined to follow Go's stdlib here and spell it `Marshaler`.

c.f. `encoding.BinaryMarshaler`.